lib/dry/system/component.rb

```diff
@@ -1,49 +1,78 @@
-require 'inflecto'
+require 'dry-equalizer'
+require 'dry/system/loader'
 
 module Dry
   module System
     class Component
-      attr_reader :loader
-      attr_reader :identifier, :path, :file
+      include Dry::Equalizer(:identifier, :path)
 
-      def initialize(loader, input)
-        @loader = loader
+      PATH_SEPARATOR = '/'.freeze
+      WORD_REGEX = /\w+/.freeze
 
-        @identifier = input.to_s.gsub(loader.path_separator, loader.namespace_separator)
+      attr_reader :identifier, :path, :file, :options, :loader
 
-        if loader.default_namespace
-          re = /^#{Regexp.escape(loader.default_namespace)}#{Regexp.escape(loader.namespace_separator)}/
-          @identifier = @identifier.sub(re, '')
-        end
+      def self.new(name, options)
+        ns, sep = options.values_at(:namespace, :separator).map(&:to_s)
 
-        @path = input.to_s.gsub(loader.namespace_separator, loader.path_separator)
-        @file = "#{path}.rb"
+        identifier = name.to_s.scan(WORD_REGEX).reject { |s| ns == s }.join(sep)
```

Comment:
It's not as robust as it could be, but in general this creates a component identifier just like it did before. ie it will do `foo/bar/` => `foo.bar` and if `:foo` is passed as the ns, then it'll end up with `bar`.